### PR TITLE
Create DM3D-SF.kicad_mod

### PR DESCRIPTION
**Pull request by savenlid**
micro sd connector

### CODE REVIEW

**Comment by evanshultz** [comment]
@SchrodingersGat @jkriege2 @poeschlr Could someone close and re-open this PR to trigger Travis?

Series datasheet: https://www.hirose.com/product/en/download_file/key_name/DM3/category/Catalog/doc_file_id/49662/?file_category_id=4&item_id=195&is_series=1

I don't see this datasheet on Hirose's site (they have only DM3A, DM3B, and DM3C), but it is in stock online and [here is the datasheet from Digi-Key](https://media.digikey.com/PDF/Data%20Sheets/Hirose%20PDFs/DM3D-SF.pdf).

Let's start with the output from KLC script:

> Checking footprint 'DM3D-SF':
>   Violating Rule 10.3
>     All other properties are left to default values. (Move and Place: Free; Auto Place: 0 and 0,  Local
>     Clearance Values: 0)
>     Attribute solder_mask_margin == 0.1 != 0!
>     Attribute solder_paste_margin == -0.05 != 0!
>   Violating Rule 10.4
>     3D model reference
>     No 3D model provided
>   Violating Rule 7.4
>     Fabrication layer requirements
>     Value Label Errors
>      - Component value is hidden (should be set to visible)
>      - Value label has a height of 0.5mm (expected: 1.0mm)
>      - Value label has a width of 0.5mm (expected: 1.0mm)
>      - Value label has a thickness of 0.1mm (expected: 0.15mm)
>     Some fabrication layer lines have a width outside allowed range of [0.1mm - 0.15mm]
>      - Line (-6.35,-10.2) -> (6.4,-10.2) on layer 'F.Fab' has width '0.05'
>      - Line (-6.35,1.5) -> (-6.35,-10.2) on layer 'F.Fab' has width '0.05'
>      - Line (6.4,1.5) -> (6.4,-10.2) on layer 'F.Fab' has width '0.05'
>   Violating Rule 8.1
>     For surface-mount devices, placement type must be set to "Surface Mount"
>     Surface Mount attribute not set
>      - For SMD footprints, 'Placement type' must be set to 'Surface mount'
>   Violating Rule 9.2
>     For through-hole components, footprint anchor is set on pad 1
>     Pad 1 not located at origin
>      - Set origin to location of Pad 1

**Comment by evanshultz** [comment]
Image:
image

This part has some area which should have a Cu keepout (note 3 in the datasheet). There are lines on the fab layer for this footprint, but that's not very descriptive for KiCad users. Have we found a way to do this in footprints? @SchrodingersGat @jkriege2 @poeschlr

Other notes:
- Footprint name should be on F.Fab layer, 1x1mm in size, and placed below the symbol
- Ref des on F.Fab should be in the center of the footprint (or at least closer considering the keepout)
- The zig zag lines in the fab layer to designate the keepout area might need to change (see above)
- The footprint origin isn't in a place which makes it easy to check against the datasheet dimensions
- Courtyard is not placed outside the component/footprint extents
- What schematic symbol in the library, if any, does this go to? I'd like to check the pin mapping.
- What is the width of pins 1-8? They're 0.55mm and I don't see it in the datasheet (surely my fault).
- Pin A is placed at -9.8mm in the Y direction. I calculate ((10.7 - 9.15) / 2) + 9.15= 9.925mm.
- F.Fab can be simplified to a 11.95x11.45mm box. The SD card placement can be optional here too.
- Silk will then be placed outside the 11.95x11.45mm box, avoiding pads.

@savenlid Will you be able to fix this footprint to meet KLC? I verified all pins look like they're placed correctly, with the exception of pin 11. Instead of further review now, I'd like to address the items noted above and by the Python scripts.

**Comment by savenlid** [comment]
Yes, this footprint may include some errors compared to datasheet.
One png attached to explain some things/errors in this footprint.
1: X origin ( I agree pin 8 would be more optimal for following the datasheet)
2: Y origin
3: width of pins 1-8 ( looks like 0.7 in datasheet so 1 Error in my footprint here, corrected)
4: Symbol, attached lib.
5: Pin A, you are correct, error corrected.
6: was marked as trough hole, now corrected to SMD
7: very good to be able to see card-protrusion, will not remove
8: Silk removed, its ju
[DM3D-SF.zip](https://github.com/KiCad/Connectors_Hirose.pretty/files/1101819/DM3D-SF.zip)
[Hirose.zip](https://github.com/KiCad/Connectors_Hirose.pretty/files/1101824/Hirose.zip)

st of no value anyway
9: Pad 11 is OK i think, both size and position
10: I have asked Hirose for step file.
11: some odd F.Fab lines removed.
12: past and mask clearance set to 0

Maybe more issues
[DM3_catalog.pdf](https://github.com/KiCad/Connectors_Hirose.pretty/files/1101794/DM3_catalog.pdf)

ps
court-yard do not always need to fend of pads, pads fend for themselves, in cases where pads is the last outpost courtyard could just go directly outside, or even in cases behind pads, as pads will be covered in electrical DRC anyway. Excessive courtyard restricts layout for physical no reason.
ds

**Comment by savenlid** [comment]
3D model attached

matches footprint nicely.
[DM3D-SF.zip](https://github.com/KiCad/Connectors_Hirose.pretty/files/1102330/DM3D-SF.zip)

**Comment by savenlid** [comment]
[DM3D-SF.zip](https://github.com/KiCad/Connectors_Hirose.pretty/files/1102339/DM3D-SF.zip)

sorry, this one is aligned.

**Comment by evanshultz** [comment]
@savenlid You will need to update your patch rather than attaching the footprint. I also don't see any PNG file attached. Once you do this, I'll go through each item. I would like if you can include pictures and more detailed comments where warranted as what I see above from you is hard to follow.

One point that does need to be made clear is that you will need to conform to all KLC rules, unless a deviation is agreed upon, in order to merge. If you are not going to meet all KLC rules then there's little point in submitting any PRs. Since you've created footprints and shown initiative to submit them here, I'd love to see your contributions in the KiCad library. A positive attitude and willingness to contribute helpfully and constructively is much appreciated.
